### ISSUE TITLE
Should we implement "updateOrInsert" method?

### ISSUE THREAD

**Issue by rakibtg** [rakibtg/SleekDB#162]
The method `updateOrInsert` will insert a new item if it's not already available otherwise update the existing item.

**Comment by Timu57** [comment]
It can use the same implementation as the `Store->update()` method, except the check if a file exists.

**Comment by REDAL** [comment]
it will be a good addon

**Comment by Timu57** [comment]
The updateOrInsert method is implemented, but I think about removing it before the next release and instead adding a new optional parameter to the `insert` and `insertMany` method.

### Current
```php
function insert(array $data): array
```

```php
function insertMany(array $data): array
```

### After
```php
function insert(array $data, bool $safeInsert = true): array
```

```php
function insertMany(array $data, bool $safeInsert = true): array
```

### $safeInsert

Can be `true` (default) or `false`.
If it is `true` the methods will **throw an exception** if a document to insert **has the _id key**.
If it is `false` the documents to insert **can have the _id key** and will **override existing documents**.

**Comment by rakibtg** [comment]
@Timu57 Thanks for the update! It's indeed a good idea. Although I think that having the `updateOrInsert` method would more easy to understand rather than using the `safeInsert` argument. It's just my opinion so you can take the decision :)

**Comment by Timu57** [comment]
I have a important question regarding the implementation.

If someone use that new method `updateOrInsert` and wants to insert the following two documents.

```php
$document1 = [
  "_id": 23,
  "name": "Tom"
];
$document2 = [
  "_id": 105,
  "name": "Max"
];
```

We have the following scenario:

`_id = 23` exist, the stored document will be overwritten.

`_id = 105` does not exist, the last stored `_id` is 67.
Should this document now be inserted with the `_id = 68` or with the `_id = 105`?

**Comment by rakibtg** [comment]
@Timu57 Is `_id` property is mandatory? It should not be a mandatory field, and if the item is new then we should ignore the `_id` due to this implementation of generating unique ID for each documents
https://github.com/rakibtg/SleekDB/blob/48d962644d4d93b38a0918d002f525d2f6e7954c/src/Store.php#L711-L720

**Comment by Timu57** [comment]
@rakibtg No, `_id` is not mandatory. That was an example if the user wants to use `updateOrInsert` with that documents..
Ok, then I will change the implementation of `updateOrInsert`.
Id `_id` does not exist in the database, it will be ignored and the next `_id` will be generated automatically.

Then I will implement the $safeInsert for `insert` and `insertMany` methods, with that the user can insert documents with the given `_id` (No auto-generation)

**Comment by Timu57** [comment]
Implemented with version 2.9
